### PR TITLE
Multi-label metrics

### PR DESCRIPTION
**Pull request by ybubnov**
This patch defined additional parameters of the metrics to evaluate certain class.

references #3.

### CODE REVIEW

**Comment by ybubnov** [comment]
@Avcu, hi I think you'll entertain this solution: you could specify a certain label to the metric itself:
```
# For binary crossentropy with 2-label output.
precision = keras_metrics.precision(label=1)
```

**Comment by Avcu** [comment]
Hi, thanks for the quick replies. I agree rather than making it explicitly, using label numbers is much more professional. However, to the best of my knowledge, the recall and precision are very special metrics unlike accuracy. They're specifically defined for the binary classification problems. For more information please refer to wiki page below. 
Shortly, they're not used for multi-classification problem unless you want to define new metrics, for example one can create 3 binary classification problem out of multi-classification problem consisting of 3 classes. (Class 1 as '1' and Class 2&Class 3 are '0', Class 2 as '1' and Class1&Class3 as '0' and so on...) I guess, that's the only case in which recall and precision make sense to use.
Secondly, I think it's wrong to say 'recall for the label' or 'precision for the second label' as you can see from the page below, they're strictly defined already. 

What we should is that regardless of number of units at the last layer(indeed it can be either 1 or 2 for the binary classification and note that using 2 units is often suggested for a better convergence), we should be always taking the predictions and true labels correctly. So, I don't see any advantage in giving the ability to choose layer to the users. But we can implement what I've done more professional.
Sorry for the long post, hope I am clear.  

https://en.wikipedia.org/wiki/Precision_and_recall

File: README.md [hunk 1]
```diff
@@ -29,4 +29,24 @@ model.compile(optimizer="sgd",
...
+model = models.Sequential()
+model.add(keras.layers.Dense(1, activation="sigmoid", input_dim=2))
+model.add(keras.layers.Dense(2, activation="softmax"))
+
+# Calculate precision for the second label.
+precision = keras_metrics.precision(label=1)
+
+# Calculate recall for the first label.
+recall = keras_metrics.precision(label=0)
+
```
Comment (Avcu) [inline README.md]: We should enable(limit indeed) users to use precision and recall without defining label
```
model = models.Sequential()
model.add(keras.layers.Dense(10, activation="sigmoid", input_dim=2))
model.add(keras.layers.Dense(2, activation="softmax"))
precision = keras_metrics.precision()
recall = keras_metrics.precision()
```
For the multi-classification, on the other hand layer should be indicated
```
model2 = models.Sequential()
model2.add(keras.layers.Dense(10, activation="sigmoid", input_dim=2))
model2.add(keras.layers.Dense(3, activation="softmax"))
precision2 = keras_metrics.precision(label=1)
recall2 = keras_metrics.precision(label=1)
```

File: keras_metrics/metrics.py [hunk 2]
```diff
@@ -1,20 +1,43 @@
...
+
+    def _binary(self, y_true, y_pred, dtype, label=0):
+        column = slice(label, label+1)
+
+        # Slice a column of the output array.
+        y_true = y_true[:,column]
+        y_pred = y_pred[:,column]
+        return self._categorical(y_true, y_pred, dtype)
+
+    def _categorical(self, y_true, y_pred, dtype):
```
Comment (Avcu) [inline keras_metrics/metrics.py]: Can you check the number of layer here, if it is bigger than 2, error should be given to user to make him/her choose layer

File: keras_metrics/metrics.py [hunk 3]
```diff
@@ -1,20 +1,43 @@
...
+        y_true = K.cast(y_true, dtype)
+        y_pred = K.cast(K.round(y_pred), dtype)
+        return y_true, y_pred
+
```
Comment (Avcu) [inline keras_metrics/metrics.py]: Can you check the number of layer here inside the method, if it is bigger than 2, error should be given to user to make him/her choose layer

File: README.md (hunk 1 above)
Comment (ybubnov) [reply README.md]: Parameter ```label``` is optional, if it's not specified, metric works like before, so the changes are backward-compatible.

Comment (Avcu) [reply README.md]: That's what exactly I meant indeed if someone puts two units without labeling, the answer will be wrong(because layer has to be equal to 1 for two units). But it is matter of coding actually and I am not an expert to be honest. Just go over the options for you to decide
1-) Merged as it is and give the responsibility to the user.
2-) For one unit and two units, it is definitely clear which layer to choose. And 99% of the people who use this library will be having one or two layers. So you can offer them a library that choose the correct layer to calculate precision and recall for one unit and for two units.

**Comment by ybubnov** [comment]
@Avcu, fixed the implementation to make the library more user-friendly.

**Comment by Avcu** [comment]
@ybubnov looks perfect
